### ISSUE TITLE
Task startup / teardown code

### ISSUE THREAD

**Issue by vasslitvinov** [chapel-lang/chapel#9747]
As an application developer, I would like to be able to request that certain code be run at the start or at the end of each task spawned by the parallel iterator of my forall loop. In some cases I would like to restrict this code to run only once per locale, regardless of how many tasks are spawned on that locale by the parallel iterator.

Acceptance criteria: the compiler accepts and implements properly forall loops with specified task startup and teardown code, both when labeled for "once per locale" or not.

**Comment by vasslitvinov** [comment]
Closing this, as it is a duplicate of #6334 .
